### PR TITLE
[INJIMOB-1854] add validation for issuer wellknown response

### PR DESCRIPTION
**Pull request by abhip2565**

### CODE REVIEW

File: src/main/java/io/mosip/mimoto/util/Utilities.java [hunk 1]
```diff
@@ -78,14 +78,14 @@ public class Utilities {
...
 //    uncomment for running mimoto Locally to populate the issuers json
-//    public Utilities(@Value("classpath:mimoto-issuers-config.json") Resource resource,
-//                     @Value("classpath:mimoto-trusted-verifiers.json") Resource trustedVerifiersResource,
-//                     @Value("classpath:/templates/credential-template.html") Resource credentialTemplateResource) throws IOException{
-//
-//        issuersConfigJsonString = (Files.readString(resource.getFile().toPath()));
-//        trustedVerifiersJsonString = (Files.readString(trustedVerifiersResource.getFile().toPath()));
-//        credentialTemplateHtmlString = (Files.readString(credentialTemplateResource.getFile().toPath()));
-//    }
+    public Utilities(@Value("classpath:mimoto-issuers-config.json") Resource resource,
```
Comment (swatigoel) [inline src/main/java/io/mosip/mimoto/util/Utilities.java:81]: revert this change

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package io.mosip.mimoto.exception;
+
+public class InvalidWellknownResponseException extends BaseCheckedException {
+
+    public InvalidWellknownResponseException() {
+        super(PlatformErrorMessages.INVALID_CREDENTIAL_ISSUER_WELLKNOWN_RESPONSE_EXCEPTION.getCode(),PlatformErrorMessages.INVALID_CREDENTIAL_ISSUER_WELLKNOWN_RESPONSE_EXCEPTION.getMessage());
+    }
+
+    public InvalidWellknownResponseException(String message) {
+        super(PlatformErrorMessages.INVALID_CREDENTIAL_ISSUER_WELLKNOWN_RESPONSE_EXCEPTION.getCode(),PlatformErrorMessages.INVALID_CREDENTIAL_ISSUER_WELLKNOWN_RESPONSE_EXCEPTION.getMessage()+"\n"+message);
```
Comment (vijay151096) [inline src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java:10]: why are we appending the error message with "/n" when this is thrown to front end , it might have UI issues if directly displayed from UI. shall we remove it.

File: src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    }
+
+    public InvalidWellknownResponseException(String message, Throwable cause) {
+        super(PlatformErrorMessages.INVALID_CREDENTIAL_ISSUER_WELLKNOWN_RESPONSE_EXCEPTION.getCode(),message, cause);
+    }
+
+    protected InvalidWellknownResponseException( Throwable cause) {
```
Comment (vijay151096) [inline src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java:18]: are we using this contructor, if not we shall remove it

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    private Validator validator;
+
+    @BeforeEach
+    void setUp() {
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenCredentialIssuerIsMissingInCredentialIssuerWellknownResponse() throws ApiNotAccessibleException {
```
Comment (vijay151096) [inline src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:39]: can we follow the AAA rule in the test, and have clear separation between all A's

and if we are expecting throws 
mention it like this in the annotation

@Test(expected = Exception.class)

this way its more clear

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        CredentialIssuerWellKnownResponse response = getCredentialIssuerWellKnownResponseDto("Issuer1",
+                Map.of("CredentialType1", getCredentialSupportedResponse("CredentialType1")));
+
+        response.setCredentialIssuer("https://valid.url/credential");
+        response.setAuthorizationServers(Collections.emptyList());  // Invalid empty list
+
+        CredentialIssuerWellknownResponseValidator credentialIssuerWellknownResponseValidator = new CredentialIssuerWellknownResponseValidator();
+        InvalidWellknownResponseException invalidWellknownResponseException = assertThrows(InvalidWellknownResponseException.class, () ->
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertEquals("RESIDENT-APP-041", invalidWellknownResponseException.getErrorCode());
```
Comment (vijay151096) [inline src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:78]: Can we import Assertions and access the assertEquals directly

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("proofSigningAlgValuesSupported: must not be null"));
+    }
+
+    @Test
+    public void shouldDetectMissingMandatoryFieldsOfCredentialSupportedDisplayResponse() {
+        // Create a new response object with nested display response details
+        CredentialIssuerWellKnownResponse response = getCredentialIssuerWellKnownResponseDto("Issuer1",
+                Map.of("CredentialType1", getCredentialSupportedResponse("CredentialType1")));
+        List<CredentialSupportedDisplayResponse> credentialSupportedDisplayResponseList = new ArrayList<>();
```
Comment (vijay151096) [inline src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:195]: Lets Extract the object creation to TestUtilities.

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java (hunk 4 above)
Comment (swatigoel) [reply src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:39]: with @test(expected) annotation, we can't assert on error messages sent along

File: src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java (hunk 2 above)
Comment (abhip2565) [reply src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java:10]: I don't think we are throwing this to frontend. this is for mimoto logging purposes

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java (hunk 6 above)
Comment (abhip2565) [reply src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:195]: I used the base objects that were presnt in test utilities and modified the key value accordingly, to handle different scenarios

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java (hunk 5 above)
Comment (swatigoel) [reply src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:78]: yes, @abhip2565 let's import Assertions

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java (hunk 4 above)
Comment (swatigoel) [reply src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java:39]: @vijay151096 before JUnit5, we used to assert exception with annotation. But with latest one, we can assert error code, messages using assertThrows method.

File: src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java (hunk 3 above)
Comment (abhip2565) [reply src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java]: removed

File: src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java (hunk 5 above)
Comment (abhip2565) [reply src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java]: done

File: src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java (hunk 2 above)
Comment (vijay151096) [reply src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java:10]: yeah i get that, usually error message is intended for frontend.

Comment (abhip2565) [reply src/main/java/io/mosip/mimoto/exception/InvalidWellknownResponseException.java:10]: yes. We added detailed logging to figure out the actual failure scenario , in mimoto logs. 👍

**Review by swatigoel — APPROVED** [review]
